### PR TITLE
Add aarch64 testing via CirrusCI

### PR DESCRIPTION
**Pull request by IAlibay**
Partly towards #4054

Changes made in this Pull Request:
 - osx-arm64 updates:
   - use pytest-xdist
   - switch over to ventura from monterey
   - reduced CI execution time from 9m to 3m
 - aarch64 linux testing
   - 8 core, 8 GB aarch64 runner using latest python docker build
 - start using starlark config for cirrus

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

----
:books: Documentation preview :books:: https://readthedocs-preview--4120.org.readthedocs.build/en/4120/

### CODE REVIEW

**Comment by IAlibay** [comment]
9m to 3m for the cirrus osx-arm64 runner.. not bad

**Comment by IAlibay** [comment]
How interesting, we can now reproduce https://github.com/MDAnalysis/mdanalysis/issues/4112

**Comment by IAlibay** [comment]
> How interesting, we can now reproduce #4112

The issue is very much "because root", the question now is - how do we deal with this, either we detect root uuid in tests (flaky), or we just say "if you're running as a privileged user, don't".

**Comment by IAlibay** [comment]
Ok this should be good to review, will need this merged ahead of fully fixing #4054 for the v2.5.0 release.

**Comment by IAlibay** [comment]
CI issues seem to be related to pytest 7.3, not the changes added here (surprisly we touched the same tests at the same time, but it seems to be a coincidence). See: #4121

**Comment by IAlibay** [comment]
@MDAnalysis/coredevs this needs to be the next priority review so that we can get https://github.com/MDAnalysis/mdanalysis/issues/4054 done in time for the 2.5 release.

File: .cirrus.star [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# See https://cirrus-ci.org/guide/programming-tasks/ for more information on
+# starlark CirrusCI files
+# Inspired by scipy's .cirrus.star script
```
Comment (hmacdope) [inline .cirrus.star]: Link?

**Review by hmacdope — APPROVED** [review]
Nitpick but otherwise looks good, feel free to ignore.

**Pull request merged**